pkg/runtime/requestDataBuilder.go

```diff
@@ -24,7 +24,7 @@ import (
 const destinationServiceAttrName = "destination.service"
 
 func newContextWithRequestData(ctx context.Context, requestBag attribute.Bag) context.Context {
-	reqData := adapter.RequestData{}
+	reqData := &adapter.RequestData{}
 	if requestBag != nil {
```

Comment:
When would requetBag be nil?

It'd be better if we can guarantee that there's a destination service present so that adapters don't need to worry about it.

Reply:
requestBag is nil in our unit tests :-), some of the unit tests don't care about passing requestBag causing the test to fail due to NPE.
With the current code, if the destination service is  not present, the RequestData.Destination.FullName will be empty string, this is the best we can do in the cases where the requestBag does not have destination.service in them.

Thoughts ?

Reply:
I'd rather not support the case where there isn't a destination service attribute. This is not behavior we want at runtime. Can we fix the tests to pass this attribute in systematically?